src/lightning_app/utilities/signals.py

```diff
@@ -0,0 +1,25 @@
+class SignalHandlerCompose:
+
+    """This class enables to compose dynamically signal handlers.
+
+    Example:
+        import signal
+
+        signal.signal = SignalHandlerCompose(signal.signal)
+    """
+
+    def __init__(self, original_signal):
+        self.original_signal = original_signal
+        self.signums = {}
+
+    def run(self, signum, frame):
+        for callable in self.signums[signum]:
+            callable(signum, frame)
+
+    def __call__(self, signum, callable):
+        if signum not in self.signums:
+            self.signums[signum] = []
+
+        self.signums[signum].append(callable)
+
+        self.original_signal(signum, self.run)
```

Comment:
we shouldn't do this for each signal registration of same `signum`

Reply:
Also, if there are signals registered already, we shouldn't be overwriting those. Instead we could get it using `getsignal(signum)` and keep it at the top of the stack